project_generation/content/templates/api/service/service.go.tmpl

```diff
@@ -61,32 +61,67 @@ func Run(buildTime, gitCommit, version string, svcErrors chan error) (*Service,
 	return &Service{
 		Config:      cfg,
 		Router:      r,
-		API:         a,
-		HealthCheck: &hc,
-		server:      s,
+		Api:         a,
+		HealthCheck: hc,
+		ServiceList: serviceList,
+		Server:      s,
 	}, nil
 }
 
-// Gracefully shutdown the service
-func (svc *Service) Close(ctx context.Context) {
+// Close gracefully shuts the service down in the required order, with timeout
+func (svc *Service) Close(ctx context.Context) error {
 	timeout := svc.Config.GracefulShutdownTimeout
 	log.Event(ctx, "commencing graceful shutdown", log.Data{"graceful_shutdown_timeout": timeout}, log.INFO)
 	ctx, cancel := context.WithTimeout(ctx, timeout)
-	defer cancel()
 
-	// stop any incoming requests before closing any outbound connections
-	if err := svc.server.Shutdown(ctx); err != nil {
-		log.Event(ctx, "failed to shutdown http server", log.Error(err), log.ERROR)
+	// track shutown gracefully closes up
+	var hasShutdownError bool
+
+	go func() {
+		defer cancel()
+
+		// stop healthcheck, as it depends on everything else
+		if svc.ServiceList.HealthCheck {
+			svc.HealthCheck.Stop()
+		}
+
+		// stop any incoming requests before closing any outbound connections
+		if err := svc.Server.Shutdown(ctx); err != nil {
+			log.Event(ctx, "failed to shutdown http server", log.Error(err), log.ERROR)
+			hasShutdownError = true
+		}
+
```

Comment:
I think we should have a comment here, like:
```
ADD CODE HERE: Close other dependencies, in the expected order
```
or something similar.

Reply:
ok, done